Sources/NIO/ByteBuffer-aux.swift

```diff
@@ -373,6 +373,22 @@ extension ByteBuffer {
         self._moveWriterIndex(forwardBy: written)
         return written
     }
+    
+    /// Writes `byte` `repeated` times. Moves the writer index forward by the number of bytes written.
+    ///
+    /// - parameter byte: The `UInt8` byte to repeat.
+    /// - parameter count: How many times to repeat the given `byte`
+    /// - returns: How many bytes were written.
+    @discardableResult
+    @inlinable
+    public mutating func writeRepeatingByte(_ byte: UInt8, count: Int) -> Int {
+        let pointer = malloc(count)
+        memset(pointer!, Int32(byte), count)
```

Comment:
You’re allocating memory here, then setting then bytes, and then copying them all into the target buffer. We can skip the first and the third operation (allocation & copy) and write straight into the target.

Check out [`withUnsafeWritableBytes(minimumWritableBytes:)`](https://apple.github.io/swift-nio/docs/current/NIO/Structs/ByteBuffer.html#/s:3NIO10ByteBufferV27writeWithUnsafeMutableBytes015minimumWritableH0_S2i_SiSwKXEtKF)
M

Reply:
Actually, [`writeWithUnsafeWritableBytes(minimumWritableBytes:)`](https://apple.github.io/swift-nio/docs/current/NIO/Structs/ByteBuffer.html#/s:3NIO10ByteBufferV27writeWithUnsafeMutableBytes015minimumWritableH0_S2i_SiSwKXEtKF) is even better, that moves the writer index for you too.

Reply:
Thanks a lot - I was looking for a function like that but clearly missed it

Reply:
As for the leak - oops, just idiocy on my part